addon/globalPlugins/pcKbBrl.py

```diff
@@ -367,15 +411,111 @@ def makeSettings(self, settingsSizer):
+		try:
+			self.dot2KeysEdit.SetValue(config.conf["pcKbBrl"]["dot2"])
+		except KeyError:
+			pass
+		# Translators: label of a dialog.
+		dot3KeysLabel = _("Type the characters for dot3 when writing with one hand.")
+		self.dot3KeysEdit = sHelper.addLabeledControl(dot3KeysLabel, wx.TextCtrl)
+		try:
+			self.dot3KeysEdit.SetValue(config.conf["pcKbBrl"]["dot3"])
+		except KeyError:
+			pass
+		# Translators: label of a dialog.
+		dot4KeysLabel = _("Type the characters for dot4 when writing with one hand.")
+		self.dot4KeysEdit = sHelper.addLabeledControl(dot4KeysLabel, wx.TextCtrl)
+		try:
+			self.dot4KeysEdit.SetValue(config.conf["pcKbBrl"]["dot4"])
+		except KeyError:
+			pass
+		# Translators: label of a dialog.
+		dot5KeysLabel = _("Type the characters for dot5 when writing with one hand.")
+		self.dot5KeysEdit = sHelper.addLabeledControl(dot5KeysLabel, wx.TextCtrl)
+		try:
+			self.dot5KeysEdit.SetValue(config.conf["pcKbBrl"]["dot5"])
+		except KeyError:
+			pass
+		# Translators: label of a dialog.
+		dot6KeysLabel = _("Type the characters for dot6 when writing with one hand.")
+		self.dot6KeysEdit = sHelper.addLabeledControl(dot6KeysLabel, wx.TextCtrl)
+		try:
+			self.dot6KeysEdit.SetValue(config.conf["pcKbBrl"]["dot6"])
+		except KeyError:
+			pass
+		# Translators: label of a dialog.
+		dot7KeysLabel = _("Type the characters for dot7 when writing with one hand.")
+		self.dot7KeysEdit = sHelper.addLabeledControl(dot7KeysLabel, wx.TextCtrl)
+		try:
+			self.dot7KeysEdit.SetValue(config.conf["pcKbBrl"]["dot7"])
+		except KeyError:
+			pass
+		# Translators: label of a dialog.
+		dot8KeysLabel = _("Type the characters for dot8 when writing with one hand.")
+		self.dot8KeysEdit = sHelper.addLabeledControl(dot8KeysLabel, wx.TextCtrl)
+		try:
+			self.dot8KeysEdit.SetValue(config.conf["pcKbBrl"]["dot8"])
+		except KeyError:
+			pass
+		# Translators: label of a dialog.
+		nullKeysLabel = _("Type the characters to be &ignored when typing in braille.")
+		self.nullKeysEdit = sHelper.addLabeledControl(nullKeysLabel, wx.TextCtrl)
+		try:
+			self.nullKeysEdit.SetValue(config.conf["pcKbBrl"]["nullKeys"])
+		except KeyError:
+			pass
+		# Translators: label of a dialog.
+		self.restoreDefaultsButton = sHelper.addItem(wx.Button(self, label=_("&Restore defaults")))
+		self.restoreDefaultsButton.Bind(wx.EVT_BUTTON, self.onRestoreDefaults)
 
 	def postInit(self):
 		self.oneHandModeCheckBox.SetFocus()
 
+	def onRestoreDefaults(self, evt):
+		self.oneHandModeCheckBox.SetValue(False)
+		self.speakDotCheckBox.SetValue(False)
+		self.timeoutSpinControl.SetValue(0)
+		self.confirmKeysEdit.SetValue("gh")
+		self.cancelKeysEdit.SetValue("ty")
+		self.dot1KeysEdit.SetValue("")
+		self.dot2KeysEdit.SetValue("")
+		self.dot3KeysEdit.SetValue("")
+		self.dot4KeysEdit.SetValue("")
+		self.dot5KeysEdit.SetValue("")
+		self.dot6KeysEdit.SetValue("")
+		self.dot7KeysEdit.SetValue("")
+		self.dot8KeysEdit.SetValue("")
+		self.nullKeysEdit.SetValue("")
```

Comment:
Just a suggestion for maintenability: use config spec values (should be OK after config spec fix)

```suggestion
		self.oneHandModeCheckBox.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'oneHandMode']).default)
		self.speakDotCheckBox.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'speakDot']).default)
		self.timeoutSpinControl.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'timeout']).default)
		self.confirmKeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'confirmKeys']).default)
		self.cancelKeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'cancelKeys']).default)
		self.dot1KeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'dot1']).default)
		self.dot2KeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'dot2']).default)
		self.dot3KeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'dot3']).default)
		self.dot4KeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'dot4']).default)
		self.dot5KeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'dot5']).default)
		self.dot6KeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'dot6']).default)
		self.dot7KeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'dot7']).default)
		self.dot8KeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'dot8']).default)
		self.nullKeysEdit.SetValue(config.conf.getConfigValidation(['pcKbBrl', 'nullKeys']).default)
```